azimuth/dataset_split_manager.py

```diff
@@ -625,3 +626,40 @@ def load_cache(self, folder: str) -> Dataset:
 
     def _get_new_version_path(self, directory):
         return pjoin(directory, f"version_{time.time()}.arrow")
+
+    def save_proposed_actions_to_csv(self) -> str:
+        """Save proposed actions to a csv file.
+
+        Returns:
+            Path to the csv file.
+        """
+        log.info("Saving proposed actions as csv.", path=self._artifact_path)
+        timestamp = time.strftime("%Y%m%d_%H%M%S", time.localtime())
+        path = pjoin(
+            self._artifact_path,
+            f"azimuth_export_{self.config.name}_{self.name}_proposed_actions_{timestamp}.csv",
+        )
+
+        ds_with_proposed_actions = filter_dataset_split(
+            self.get_dataset_split(),
+            filters=DatasetFilters(data_action=ALL_DATA_ACTIONS),
+            config=self.config,
+        )
+
+        persistent_id = self.config.columns.persistent_id
+        df = (
+            ds_with_proposed_actions.remove_columns(
+                list(
+                    set(ds_with_proposed_actions.column_names)
+                    - set(ALL_DATA_ACTIONS + [persistent_id])
+                )
+            )
+            .to_pandas()
+            .set_index(persistent_id)
+        )
```

Comment:
Instead of creating a list, then converting it to a set, you can create a set directly:
```diff
-set(ALL_DATA_ACTIONS + [persistent_id])
+{*ALL_DATA_ACTIONS, persistent_id}
```
Also, I think the odd indent and parentheses situation happens because of `.to_pandas().set_index(persistent_id)`. If we separate them on a second line, it becomes a lot more compact:
```suggestion
        ds_with_proposed_actions = ds_with_proposed_actions.remove_columns(
            list(set(ds_with_proposed_actions.column_names) - {*ALL_DATA_ACTIONS, persistent_id})
        )
        df = ds_with_proposed_actions.to_pandas().set_index(persistent_id)
```

Reply:
Good idea!